hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/MetricAggregationExpressionConverter.java

```diff
@@ -43,35 +45,39 @@
 
   Single<Expression> convert(MetricAggregationRequest metricAggregation, String alias) {
     return this.buildAggregationFunctionExpression(
-            metricAggregation.attribute(),
+            metricAggregation.attributeExpression(),
             metricAggregation.aggregation(),
             metricAggregation.arguments(),
             alias)
         .map(functionExpression -> Expression.newBuilder().setFunction(functionExpression).build());
   }
 
   Single<Expression> convertForNoArgsOrAlias(
-      AttributeModel attribute, AttributeModelMetricAggregationType aggregationType) {
-    return convertForArgsButNoAlias(attribute, aggregationType, Collections.emptyList());
+      AttributeAssociation<AttributeExpression> attributeExpression,
+      AttributeModelMetricAggregationType aggregationType) {
+    return convertForArgsButNoAlias(attributeExpression, aggregationType, Collections.emptyList());
```

Comment:
while reading I had to look for type, so would `attributeAssociation` would have been a better argument name?
`AttributeAssociation<AttributeExpression> attributeExpression` -> `AttributeAssociation<AttributeExpression> attributeAssociation`?

If, we change, the same thing for appropriate places in other files too. If, we keep the same, it's fine too.

Reply:
yeah this naming's tricky. `AttributeAssociation` is really just a wrapper to materialize an attribute reference on another thing, attribute expression is the kernel of data that's more relevant. So I could call it `attributeExpressionAssociation` if you think that'd be clearer?

Reply:
I think, that would be better - `attributeExpressionAssociation`

Reply:
ok, that likely goes into the core code as well so will do that pass starting there.

Reply:
https://github.com/hypertrace/hypertrace-core-graphql/pull/92

Reply:
Maybe there were many files, it was really hard what is what `attributeExpressionAssociation` or `attributeExpression`, but, it more clear to me. Thanks a lot for making this change.
Reviewed - hypertrace/hypertrace-core-graphql#92

Reply:
Merged and updated in this PR